File: src/main/java/com/ptaf/ai/context/FrameworkGenerationContext.java
package com.ptaf.ai.context;

import java.util.List;

public record FrameworkGenerationContext(
        List<String> existingFeatureSnippets,
        List<String> existingStepDefinitions,
        List<String> existingYamlKeys,
        List<String> uiElementKeys,
        List<String> apiRequestKeys,
        List<String> dbQueryKeys
) {
    public int featureSnippetCount() {
        return existingFeatureSnippets != null ? existingFeatureSnippets.size() : 0;
    }

    public int stepDefinitionCount() {
        return existingStepDefinitions != null ? existingStepDefinitions.size() : 0;
    }

    public int yamlKeyCount() {
        return existingYamlKeys != null ? existingYamlKeys.size() : 0;
    }
}

File: src/main/ptaf/ai/context/framework_generation_context.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class FrameworkGenerationContext:
    existing_feature_snippets: list[str]
    existing_step_definitions: list[str]
    existing_yaml_keys: list[str]
    ui_element_keys: list[str]
    api_request_keys: list[str]
    db_query_keys: list[str]

    @property
    def feature_snippet_count(self) -> int:
        return len(self.existing_feature_snippets)

    @property
    def step_definition_count(self) -> int:
        return len(self.existing_step_definitions)

    @property
    def yaml_key_count(self) -> int:
        return len(self.existing_yaml_keys)
